examples/imagenet/input_pipeline.py

```diff
@@ -172,42 +168,50 @@ def preprocess_for_eval(image_bytes, dtype=tf.float32, image_size=IMAGE_SIZE):
   """
   image = _decode_and_center_crop(image_bytes, image_size)
   image = tf.reshape(image, [image_size, image_size, 3])
-  image = normalize_image(image)
+  image = _normalize_image(image)
   image = tf.image.convert_image_dtype(image, dtype=dtype)
   return image
 
 
-def load_split(batch_size, train, dtype=tf.float32, image_size=IMAGE_SIZE, cache=False):
+def create_split(dataset_builder: tfds.core.DatasetBuilder, batch_size: int,
+                 train: bool, dtype: tf.DType = tf.float32,
+                 image_size: int = IMAGE_SIZE, cache: bool = False):
   """Creates a split from the ImageNet dataset using TensorFlow Datasets.
 
   Args:
+    dataset_builder: TFDS dataset builder for ImageNet.
```

Comment:
Can't the dataset builder just be constructed here rather than at the top level of the training loop?

Reply:
Discussed in the previous comment. Basically to avoid multiple initializations of the dataset_builder. 

The values of number of train and eval images are required in main train loop to compute the steps per epoch and here as well to compute split. We definitely don't want to have hardcoded numbers. So initialized it in the main train loop and passing it on as argument.